src/libraries/System.Text.Json/docs/collection_support.md

```diff
@@ -0,0 +1,131 @@
+# JsonSerializer collection support in 3.1.x
+
+All collections (any type that derives from IEnumerable) are supported for
+serialization with JsonSerializer; we simply call the .GetEnumerator() method
+of the collection, and write the elements if they are supported.
+Deserialization is more complicated and is not supported for all collections.
+
+This document gives a quick overview of which collections are supported.
+
+## Built-in types
+
+### System.Collections
+
+| Type | Serialization | Deserialization |
+| --- | --- | --- |
+| ArrayList | Supported | Supported |
+| BitArray | Supported | Not supported |
+| Hashtable | Supported | Supported |
+| Queue | Supported | Supported |
+| SortedList | Supported | Supported |
+| Stack | Supported | Supported |
+| DictionaryEntry | Supported | Supported |
+| ICollection | Supported | Supported |
+| IDictionary | Supported | Supported |
+| IEnumerable | Supported | Supported |
+| IList | Supported | Supported |
+
+### System.Collections.Generic
+
+| Type | Serialization | Deserialization |
+| --- | --- | --- |
+| Dictionary<string, TValue> | Supported | Supported |
+| HashSet<T> | Supported | Supported |
+| LinkedList<T> | Supported | Supported |
+| LinkedListNode<T> | Supported | Not supported |
+| List<T> | Supported | Supported |
+| Queue<T> | Supported | Supported |
+| SortedDictionary<string,TValue> | Supported | Supported |
+| SortedList<string,TValue> | Supported | Supported |
+| SortedSet<T> | Supported | Supported |
+| Stack<T> | Supported | Supported |
+| KeyValuePair<TKey, TValue> | Supported | Supported |
+| IAsyncEnumerable<T> | Not supported | Not supported |
+| ICollection<T> | Supported | Supported |
+| IDictionary<string,TValue> | Supported | Supported |
+| IEnumerable<T> | Supported | Supported |
+| IList<T> | Supported | Supported |
+| IReadOnlyCollection<T> | Supported | Supported |
+| IReadOnlyDictionary<string,TValue> | Supported | Supported |
+| IReadOnlyList<T> | Supported | Supported |
+| ISet<T> | Supported | Supported |
+
+### System.Collections.Immutable
+
+| Type | Serialization | Deserialization |
+| --- | --- | --- |
+| ImmutableArray<T> | Supported | Supported |
+| ImmutableDictionary<string, TValue> | Supported | Supported |
+| ImmutableHashSet<T> | Supported | Supported |
+| ImmutableList<T> | Supported | Supported |
+| ImmutableQueue<T> | Supported | Supported |
+| ImmutableSortedDictionary<string, TValue> | Supported | Supported |
+| ImmutableSortedSet<T> | Supported | Supported |
+| ImmutableStack<T> | Supported | Supported |
+| IImmutableDictionary<string, TValue> | Supported | Supported |
+| IImmutableList<T> | Supported | Supported |
+| IImmutableQueue<T> | Supported | Supported |
+| IImmutableSet<T> | Supported | Supported |
+| IImmutableStack<T> | Supported | Supported |
+
+### System.Collections.Specialized
+
+#### Deserialization heuristics
```

Comment:
What does this mean?

Reply:
I had notes earlier about how the deserialization is done for each class of types. This is not relevant is this doc so I removed them (and forgot this one). I just pushed a commit to remove this line.